src/test/java/com/prgrms/springbootbasic/customer/storage/CustomerRepositoryTest.java

```diff
@@ -0,0 +1,277 @@
+    JdbcCustomerRepository jdbcCustomerRepository;
+
+    @Autowired
+    DataSource dataSource;
+
+    private Customer customer;
+    private Customer wrongNameCustomer;
+    private List<Customer> customerList;
+
+    @BeforeAll
+    public void setDatabase() {
+
+        MysqldConfig mysqldConfig = aMysqldConfig(v5_7_latest)
+                .withCharset(Charset.UTF8)
+                .withPort(2215)
+                .withUser("test", "test1234!")
+                .withTimeZone("Asia/Seoul")
+                .build();
+        anEmbeddedMysql(mysqldConfig)
+                .addSchema("test-order_mgmt", classPathScript("customer.sql"))
+                .start();
+    }
+
+    @BeforeEach
+    public void setup() {
+        customer = new Customer(UUID.randomUUID(), LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS), "test");
+        wrongNameCustomer = new Customer(UUID.randomUUID(),
+                LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS),
+                tooLongName);
+        customerList = List.of(
+                new Customer(UUID.randomUUID(), LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS), "test1"),
+                new Customer(UUID.randomUUID(), LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS), "test2"),
+                new Customer(UUID.randomUUID(), LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS), "test3")
+        );
+    }
+
+    @AfterEach
+    public void deleteAll() {
+        jdbcTemplate.update("DELETE FROM customer");
+    }
+
+    @Test
+    @DisplayName("Customer를 저장할 수 있다.")
+    void save() {
+        //given&when
+        jdbcCustomerRepository.save(customer);
+        Optional<Customer> found = jdbcCustomerRepository.findById(customer.getId());
+
+        //then
+        assertThat(found.isPresent()).isTrue();
+        Customer foundCustomer = found.get();
+
+        assertThat(foundCustomer)
+                .usingRecursiveComparison()
+                .isEqualTo(customer);
+    }
+
+    @Test
+    @DisplayName("영문 기준 50글자를 초과하면 저장에 실패한다.")
+    void saveFailStringTooLong() {
+        //when&then
+        assertThrows(DataIntegrityViolationException.class, () -> jdbcCustomerRepository.save(wrongNameCustomer));
+    }
+
+    @Test
+    @DisplayName("id를 조회해서 Customer를 조회할 수 있다.")
+    void findById() {
+        //given
+        jdbcCustomerRepository.save(customer);
+
+        //when
+        Optional<Customer> found = jdbcCustomerRepository.findById(customer.getId());
+
+        //then
+        assertThat(found.isPresent()).isTrue();
+
+        assertThat(found.get())
+                .usingRecursiveComparison()
+                .isEqualTo(customer);
+    }
+
+    @Test
+    @DisplayName("모든 Customer를 조회할 수 있다.")
+    void findAll() {
+        //given
+        customerList.forEach(jdbcCustomerRepository::save);
+
+        //when
+        List<Customer> foundCustomers = jdbcCustomerRepository.findAll();
+
+        //then
+        assertThat(customerList.size()).isEqualTo(foundCustomers.size());
```

Comment:
```suggestion
        var actual = jdbcCustomerRepository.findAll().size();
        assertThat(actual).isEqualTo(3);
```

- assertThat 첫번째인자는 실제 값이 들어가고(파라미터 힌트 참고), 뒤에들어가는 값은 예상하는 값을 적습니다.
- 또 실제값에 로직이 들어가면 읽기가 어려워요.
- 예상값(expected)은 변수 말고, 값으로 표현하면 읽기 좋더라구요. 환님은 어떠세요?

Reply:
제가 실제값이랑 예측값이랑 매번 콘솔을 봐도 햇갈렸던 이유가 이거였을까요? 🤣 기준 없이 실제 값과 예상값을 집어넣기만 했네요!

`예상값(expected)은 변수 말고, 값으로 표현하면 읽기 좋더라구요.` 에 대해서는, 사실 위에 리뷰와 같이 생각했을 때, 코드상에 foreach를 쓰지 않고 풀어쓴다음, 예상 값에 값(3)으로 표현하는게 테스트로서 가동성이 더 좋아질 것 같습니다!